tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/internal/types/TBfloat16Mapper.java

```diff
@@ -55,4 +79,49 @@ public RawTensor asRawTensor() {
       this.rawTensor = rawTensor;
     }
   }
+
+  private static final class SparseTBfloat16 extends FloatSparseNdArray implements TBfloat16, SparseTensor<TBfloat16> {
+
+    @Override
+    public Class<TBfloat16> type() {
+      return TBfloat16.class;
+    }
+
+    @Override
+    public DataType dataType() {
+      return values().dataType();
+    }
+
+    @Override
+    public long numBytes() {
+      return SparseHelpers.numBytes(this);
+    }
+
+    @Override
+    public void close() {
+      // sparse tensors do not own the dense tensors that compose them, nothing to close
```

Comment:
Since all tensors created should normally be protected by a `try-with-resource` enclosure, they should normally be released after the sparse tensor, like in this snippet: 
```java
try (TInt64 indices = TInt64.tensorOf(...);
       TFloat64 values = TFloat64.vectorOf(...);
       TInt64 denseShape = TInt64.vectorOf(...);
       TFloat64 sparseTensor = TFloat64.sparseTensorOf(indices, values, denseShape)) {
...
}
```
But I agree that should be documented somewhere. I'm not super satisfied of this solution though and I'm now thinking that we might want to use JavaCPP references count instead, which would support more scenarios and be less error prone. Wdyt?

Reply:
I think I'm fine with the current behaviour of the code, just it should be documented somewhere. I think reference counting is probably the way to go in the future, but I don't see a reason to hold this up while we do it.

Reply:
Actually, I've already got it working with reference counts. It still requires that all sub-components of a sparse tensor to be protected by `try-with-resources` block to avoid leaking, but it now allows the sparse tensor to live beyond that point. For example, we can support something like:
```java
TFloat64 createSparseTensor() {
    try (TInt64 indices = TInt64.tensorOf(...);
        TFloat64 values = TFloat64.vectorOf(...);
        TInt64 denseShape = TInt64.vectorOf(...)) {

        return TFloat64.sparseTensorOf(indices, values, denseShape);
    }
}

try (TFloat64 sparseTensor = createSparseTensor()) {
    ...
}
```
Should I go ahead with that implementation instead?

Reply:
If you've already got it working, then yes that sounds better than the current approach.